editor/src/clj/editor/editor_extensions/commands.clj

```diff
@@ -0,0 +1,129 @@
+;; Copyright 2020-2024 The Defold Foundation
+;; Copyright 2014-2020 King
+;; Copyright 2009-2014 Ragnar Svensson, Christian Murray
+;; Licensed under the Defold License version 1.0 (the "License"); you may not use
+;; this file except in compliance with the License.
+;;
+;; You may obtain a copy of the License, together with FAQs at
+;; https://www.defold.com/license
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns editor.editor-extensions.commands
```

Comment:
I moved commands-related code from `editor_extensions.clj` here. There is a logic change: action execution now happens using CompletableFutures, in a non-blocking way. Also, handler `active?` checks are now invoked in "immediate" mode that disallows launching long-running tasks (like writing to files). Which I think is reasonable — `active?` checks happen on UI thread as a response to user interactions, those should be very fast.